File: Swift/프로그래머스/2/155651. 호텔 대실/호텔 대실.swift
import Foundation

func solution(_ book_time:[[String]]) -> Int {
    let sortedBookings = book_time.map { 
        [timeToMinutes($0[0]), timeToMinutes($0[1]) + 10]
    }.sorted { $0[0] < $1[0] }
    
    var outTimeArr = [Int]()
    
    for booking in sortedBookings {
        let start = booking[0]
        let end = booking[1]
        
        outTimeArr.sort()
        
        if !outTimeArr.isEmpty && outTimeArr[0] <= start {
            outTimeArr.removeFirst()
        }
        
        outTimeArr.append(end)
    }
    
    return outTimeArr.count
}

func timeToMinutes(_ time: String) -> Int {
        let components = time.split(separator: ":")
        let hours = Int(components[0])! * 60
        let minutes = Int(components[1])!
        return hours + minutes
    }

File: Java/프로그래머스/2/155651. 호텔 대실/호텔 대실.java
import java.util.*;

class Solution {
    public int solution(String[][] book_time) {
        int[][] convertedTime = new int[book_time.length][2];
        
        for (int i = 0; i<book_time.length; i++) {
            int enter = timeToMin(book_time[i][0]);
            int out = timeToMin(book_time[i][1]) + 10;
            convertedTime[i][0] = enter;
            convertedTime[i][1] = out;
        }
        
        Arrays.sort(convertedTime, (x, y) -> x[0]-y[0]);
        
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        
        int answer = 0;
        for (int[] customer : convertedTime) {
            int enter = customer[0];
            int out = customer[1];
            
            if (!pq.isEmpty() && pq.peek() <= enter) {
                pq.poll();
            }
            pq.add(out);
        }
        
        return pq.size();
    }
    
    public static int timeToMin(String time) {
        String[] timeArr = time.split(":");
        int minTime = (Integer.parseInt(timeArr[0]) * 60) + Integer.parseInt(timeArr[1]);
        return minTime;
    }
}